test/replication.rest.test.js

```diff
@@ -0,0 +1,484 @@
+      // after each change, because it's tracking the changes too.
+      trackChanges: false,
+      // Enable remote replication in order to get remoting API metadata
+      // used by the remoting connector
+      enableRemoteReplication: true
+    });
+  }
+
+  function seedServerData(done) {
+    async.series([
+      function(next) {
+        serverApp.dataSources.db.automigrate(next);
+      },
+      function(next) {
+        ServerUser.deleteAll(next);
+      },
+      function(next) {
+        ServerUser.create([ALICE, PETER, EMERY], function(err, created) {
+          if (err) return next(err);
+          aliceId = created[0].id;
+          peterId = created[1].id;
+          next();
+        });
+      },
+      function(next) {
+        ServerUser.login(ALICE, function(err, token) {
+          if (err) return next(err);
+          aliceToken = token.id;
+
+          ServerUser.login(PETER, function(err, token) {
+            if (err) return next(err);
+            peterToken = token.id;
+
+            ServerUser.login(EMERY, function(err, token) {
+              emeryToken = token.id;
+
+              next();
+            });
+          });
+        });
+      },
+      function(next) {
+        ServerCar.create(
+          [
+            { maker: 'Ford', model: 'Mustang' },
+            { maker: 'Audi', model: 'R8' }
+          ],
+          function(err, cars) {
+            if (err) return next(err);
+            serverCars = cars.map(carToString);
+            next();
+          });
+      }
+    ], done);
+  }
+
+  function seedClientData(done) {
+    LocalUser.deleteAll(function(err) {
+      if (err) return done(err);
+      LocalCar.deleteAll(function(err) {
+        if (err) return done(err);
+        LocalCar.create(
+          [{ maker: 'Local', model: 'Custom' }],
+          function(err, cars) {
+            if (err) return done(err);
+            clientCars = cars.map(carToString);
+            done();
+          });
+      });
+    });
+  }
+
+  function setAccessToken(token) {
+    clientApp.dataSources.remote.connector.remotes.auth = {
+      bearer: new Buffer(token).toString('base64'),
+      sendImmediately: true
+    };
+  }
+
+  function expectHttpError(code, done) {
+    return function(err) {
+      if (!err) return done(new Error('The method should have failed.'));
+      expect(err).to.have.property('statusCode', code);
+      done();
+    };
+  }
+
+  function replicateServerToLocal(next) {
+    ServerUser.replicate(LocalUser, function(err, conflicts) {
+      if (err) return next(err);
+      if (conflicts.length) return next(conflictError(conflicts));
+      next();
+    });
+  }
+
+  function conflictError(conflicts) {
+    var err = new Error('Unexpected conflicts\n' +
+      conflicts.map(JSON.stringify).join('\n'));
+    err.name = 'ConflictError';
+  }
+
+  function carToString(c) {
+    return c.maker ? c.maker + ' ' + c.model : c.model;
+  }
+});
```

Comment:
Wow this is an impressive amount of testing. :+1: 

This is making it blatantly obvious that we need to move all of this into another module.